src/lib.rs

```diff
@@ -278,16 +282,56 @@ impl<T> ThinVec<T> {
         }
     }
 
+    /// Reserve capacity for at least `additional` more elements to be inserted.
+    ///
+    /// May reserve more space than requested, to avoid frequent reallocations.
+    ///
+    /// Panics if the new capacity overflows `usize`.
+    ///
+    /// Re-allocates only if `self.capacity() < self.len() + additional`.
     pub fn reserve(&mut self, additional: usize) {
-        // TODO
+        let len = self.len();
+        let old_cap = self.capacity();
+        let min_cap = len.checked_add(additional).expect("capacity overflow");
```

Comment:
This check is the reason why `reserve_one_more` was split off from `reserve`. I'd toss in a FIXME for it. Not a blocker.